src/test.ts

```diff
@@ -0,0 +1,176 @@
+  const strRegex =
+    '^((https|http|ftp|rtsp|mms)?://)' +
+    '?(([0-9a-z_!~*\'().&=+$%-]+: )?[0-9a-z_!~*\'().&=+$%-]+@)?' +
+    '(([0-9]{1,3}.){3}[0-9]{1,3}' +
+    '|' +
+    '([0-9a-z_!~*\'()-]+.)*' +
+    '([0-9a-z][0-9a-z-]{0,61})?[0-9a-z].' +
+    '[a-z]{2,6})' +
+    '(:[0-9]{1,4})?' +
+    '((/?)|' +
+    '(/[0-9a-z_!~*\'().;?:@&=+$,%#-]+)+/?)$';
+  const re = new RegExp(strRegex);
+  return re.test(url);
+}
+
+export async function test_all(argv: any) {
+  // extracting all arguments
+  const {
+    speed: _speed,
+    adminEmail,
+    mwUrl,
+    mwWikiPath,
+    mwApiPath,
+    mwRestApiPath,
+    mwModulePath,
+    mwDomain,
+    mwUsername,
+    mwPassword,
+    requestTimeout,
+    format,
+    filenamePrefix,
+    publisher: _publisher,
+    outputDirectory: _outputDirectory,
+    addNamespaces: _addNamespaces,
+    articleList: _articleList,
+    customZimFavicon,
+    optimisationCacheUrl,
+    noLocalParserFallback,
+    forceLocalParser,
+    customFlavour: customProcessorPath,
+  } = argv;
+
+  const cpuCount = os.cpus().length;
+
+  // checks for speed
+  test_speed(_speed);
+  const speed = Math.max(1, Math.round(cpuCount * (_speed || 1) * 3));
+
+  try {
+    if (optimisationCacheUrl) {
+      // Decompose the url with path and other S3 creds
+      const s3UrlObj = urlParser.parse(optimisationCacheUrl);
+      const queryReader = QueryStringParser.parse(s3UrlObj.query);
+      const s3Url = (s3UrlObj.host || '') + (s3UrlObj.pathname || '');
+      this.s3Obj = new S3(s3Url, queryReader);
+      await this.s3Obj.initialise().then(() => {
+        logger.log('Successfully logged in S3');
+      });
+    }
+  } catch (err) {
+    throw err;
+  }
+
+  // Extract S3 obj to pass to downloader class
+  const s3 = this.s3Obj ? this.s3Obj : {};
+
+  const mw = new MediaWiki({
+    getCategories: !!argv.getCategories,
+    apiPath: mwApiPath,
+    restApiPath: mwRestApiPath,
+    modulePath: mwModulePath,
+    base: mwUrl,
+    domain: mwDomain,
+    password: mwPassword,
+    username: mwUsername,
+    wikiPath: mwWikiPath,
+  });
+
+  const downloader = new Downloader({
+    mw,
+    uaString: `${config.userAgent} (${adminEmail})`,
+    speed,
+    reqTimeout: requestTimeout || config.defaults.requestTimeout,
+    noLocalParserFallback,
+    forceLocalParser,
+    optimisationCacheUrl,
+    s3,
+  });
+
+  // checks for mwUrl
+  test_url(mwUrl);
+
+  // checks for adminEmail
+  test_email(adminEmail);
+
+  // redis client check
+  // created a test redis client and then closed it.
+  test_redis(argv);
+
+  // checks for custom zim favicon
+  if (customZimFavicon) {
+    await test_customZimFavicon(customZimFavicon,downloader)
+    .catch(err=>{
+      throw err;
+    });
+  }
+}
+
+export function test_speed(_speed:any)
+{
+  if (_speed && isNaN(_speed)) {
+    throw new Error(
+      `speed is not a number, please give a number value to --speed.`
+    );
+  }
+}
+
+export function test_url(mwUrl:string)
```

Comment:
In this function and many other one, there is an incoherency between the name of the function and what is done. This fonction should be called `test_mwUrl` (then the error msg makes sense). Or we should have a `test_url` and then the error msg should be outside.